File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        preNum = nums[0]
        k = 1
        
        for n in range(1, len(nums)):
            if preNum == nums[k]:
                del nums[k]
                k -= 1
            else:
                preNum = nums[k]
            k += 1
        return k
                

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int write = 1;
        for(int i = 1; i < nums.length; i++) {
            if(nums[i] != nums[i-1]) {
                nums[write] = nums[i];
                write += 1;
            }
        }
        return write;
    }
}